app/models/user.rb

```diff
@@ -0,0 +1,16 @@
+class User < ActiveRecord::Base
+  has_many :tweets
+  validates :username, presence: true
+  validates :email, presence: true
+  validates :password, presence: true
+  has_secure_password
+
+  def slug
+    self.username.gsub(" ", "-")
+  end
+
+  def self.find_by_slug(slug)
+    self.all.find {|user| user.slug == slug}
```

Comment:
Instead of redefining your own `find_by_slug` method, you can utilize the existing ActiveRecord `find_by` method.

```ruby
def self.find_by_slug(slug)
  User.find_by(slug: slug)
end
```
This returns an ActiveRecord::Collection object, allowing you to chain additional ActiveRecord methods.